packages/medusa/src/loaders/index.ts

```diff
@@ -241,6 +241,14 @@ export default async ({
       },
     })
 
+    // Medusa app load all non legacy modules, so we need to register them in the container since they are into their own container
+    // We might decide to do it elsewhere but for now I think it is fine
+    for (const [serviceKey, moduleService] of Object.entries(modules)) {
```

Comment:
this will certainely change and we will have to rethink the registration.

Reply:
One think would maybe be to only use the registerModules, use the flag load legacy or not depending where we are using it and remove registerMedusaModule to cleanup a bit and mutualize. We can discuss that all together. In the mean time i would like to wait to not go to far before we decide and validate the approach suggested in this message and this quick fix works for now

Reply:
agreed with carlos